### ISSUE TITLE
[Feature Request] Support optuna 2.8.0

### ISSUE THREAD

**Issue by nikvaessen** [facebookresearch/hydra#1734]
# 🚀 Feature Request

I would like to make use of the `constant_liar` parameter of [TPESampler](https://optuna.readthedocs.io/en/stable/reference/generated/optuna.samplers.TPESampler.html) which was added in version [2.8.0](https://github.com/optuna/optuna/releases/tag/v2.8.0). This is helpful for preventing sampling around the same neighborhood when `n_jobs` is larger than 1.

## Motivation

Currently the optuna sweeper is restricted to version `<2.5.0` as per https://github.com/facebookresearch/hydra/blob/9e3a5261a91febb028fb16613b4bead535c08277/plugins/hydra_optuna_sweeper/setup.py#L31

## Additional context
What is the reason for requiring this specific version of Optuna?

**Comment by nikvaessen** [comment]
I see this is already being worked on https://github.com/facebookresearch/hydra/pull/1726

**Comment by omry** [comment]
I don't believe there is a plan to add functionality as a part of #1726. Once it's done feel free to send a followup pull request to add what you need.

**Comment by omry** [comment]
Closing as part of an automated issue cleanup sweep.

This request appears to be obsolete because the Optuna plugin currently requires `optuna>=2.10,<3.0`. If this is still relevant, or if I closed this by mistake, please comment here with updated details and we can reopen it.
